diff -uNr musl-patch-old/ldso/dynlink.c musl-patch-new/ldso/dynlink.c
--- musl-patch-old/ldso/dynlink.c
+++ musl-patch-new/ldso/dynlink.c
@@ -490,10 +490,18 @@
 			memcpy(reloc_addr, &(size_t){sym_val + addend}, sizeof(size_t));
 			break;
 		case REL_RELATIVE:
+#if DL_FDPIC
+			*reloc_addr = (size_t)laddr(dso, addend);
+#else
 			*reloc_addr = (size_t)base + addend;
+#endif
 			break;
 		case REL_SYM_OR_REL:
 			if (sym) *reloc_addr = sym_val + addend;
+#if DL_FDPIC
+			else *reloc_addr = (size_t)laddr(dso, addend);
+#else
 			else *reloc_addr = (size_t)base + addend;
+#endif
 			break;
 		case REL_COPY:
 			memcpy(reloc_addr, (void *)sym_val, sym->st_size);
